pm/helpers.go

```diff
@@ -7,7 +7,7 @@ import (
 	ethcommon "github.com/ethereum/go-ethereum/common"
 )
 
-func randHashOrFatal(t *testing.T) ethcommon.Hash {
+func RandHashOrFatal(t *testing.T) ethcommon.Hash {
```

Comment:
Feels like these functions should actually live in a common test utils file since they aren't really specific to `pm`. Unfortunately, I don't think we can put them in the `common` package right now since this would introduce a cycle (`pm` tests import `common` and `db.go`/`db_test.go` import `pm`)...Perhaps if we move the DB code into its own package then we can move these functions into the `common` package - I don't think we have to do this in this PR though

Reply:
Yes, that was exactly my thinking - that all these test helpers should go under `common` but to do that we must solve the cyclical dependency issue.

Another approach we could take is have a package dedicated to test helpers we could name `testutils` or something similar.

Wdyt?

Reply:
Rather than have `common` import `pm`, we can have `common` export a DBTicket which is converted into a `pm.Ticket` somewhere else, eg in the PM code. That would allow us to keep common helpers in `common` and avoid an import cycle.

Reply:
In the case of tickets, there is an additional annoyance in that `pm.Recipient.ReceiveTicket()` will use the method `StoreWinningTicket(string, *Ticket, []byte, *big.Int)` defined in the `TicketStore` interface to persist winning tickets. `common.DB` currently implements `TicketStore`. We can avoid importing `pm` in `common` by using `common.DBTicket` in the `TicketStore` interface: `StoreWinningTicket(string, *common.DBTicket, []byte, *big.Int)`. But, now we're importing code specific to `common.DB` into `pm` when it would be preferable to keep `pm` agnostic to how persistence is handled. 

One solution is to destructure the ticket fields in the `TicketStore` method call:

`StoreWinningTicket(string, ethcommon.Address, ethcommon.Address ... rest of ticket fields ..., []byte, *big.Int)`

Neither package would have to import the either. But this leads to a very unwieldy method call with many parameters - not great.

Alternatively, we could use an anonymous struct - the method signature would look ugly, but the method calls would be cleaner.

Another solution is to move the `Ticket` struct into the `common` package (or whatever the name is for the package we use to store common types used throughout go-livepeer). But this creates more external dependencies for the `pm` package when it would preferable to keep it self contained.

Extracting `DB` into its own package that imports `pm` and having the `common` package specifically focused on just helper functions seems cleaner than the above solutions.

Reply:
> But, now we're importing code specific to common.DB into pm when it would be preferable to keep pm agnostic to how persistence is handled.

The PM code right now is quite nicely cleaved with minimal glue needed between the DB and PM itself. However, I do wonder how hard we'll need to work to maintain this clean separation in the future.

One difficulty with the current approach is that nearly all modules could have a legitimate reason to use the DB [1], but if the DB module imports everything, none of those modules can actually invoke the DB itself. That means we need a third module to glue together the DB and the module, or to implement more of the "glue code" within the DB module itself, which bleeds over the concerns.

We could define a generic `Ticket` as an interface to avoid embedding any DB-specific constructs within the `TicketStore` interface (perhaps with a single PMTicketToGenericTicket function for the `Ticket` interface). However, I don't think there is any harm in putting DB-aware code within the PM implementation itself, as long as the interface remains storage-agnostic. We may eventually find it difficult to keep the concrete implementation completely agnostic to the DB.

[1] That most modules could make use of the DB is one reason I like to consider the DB common code, and by keeping it narrowly scoped to processing queries (eg, minimal glue), there is less reason to move the DB into its own module.

> Another solution is to move the Ticket struct into the common package (or whatever the name is for the package we use to store common types used throughout go-livepeer). But this creates more external dependencies for the pm package when it would preferable to keep it self contained.

Presumably `pm` would still be importing common helpers so it wouldn't be a technical obstacle to defining the Ticket struct within `common`. However, I understand the incongruity of separating the definition from the actual use.